pkg/api/v1/cluster_types.go

```diff
@@ -77,10 +81,12 @@ const (
 type GenericUpgradeSpec struct {
 	// FailureStrategy specifies which logic is executed when upgrade failure happens.
 	// Currently only Retry is supported.
+	// +kubebuilder:default:="Retry"
 	FailureStrategy GenericUpgradeFailureStrategy `json:"failureStrategy,omitempty"`
 	// PollInterval specifies how often upgrade logic polls on state updates.
 	// Increasing this value should lower number of requests sent to apiserver, but it may affect
 	// overall time spent during upgrade.
+	// +kubebuilder:default:="1s"
```

Comment:
This and onward should have `+optional` when omit empty https://github.com/kubernetes/community/blob/86550d6/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required